src/main/java/org/elasticsearch/river/kafka/KafkaWorker.java

```diff
@@ -100,22 +97,10 @@ private void consumeMessagesAndAddToBulkProcessor(final KafkaStream stream) {
                 final MessageAndMetadata messageAndMetadata = consumerIterator.next();
                 logMessage(messageAndMetadata);
 
-                messageSet.add(messageAndMetadata);
```

Comment:
@schonfeld, thanks for pointing this out. I just checked it locally, without having to create a set, and it works like a charm. Apparently the bug related to Bulk Processor not processing the messages has been fixed. Also, when some messages are not sent to EL yet, because the size of bulk messages has not been reached, and EL is stopped, they automatically get flushed.

See the latest version on master.